tests/contrib/io/azure/test_json_blob.py

```diff
@@ -0,0 +1,184 @@
+    try:
+        blob_json_data_set().load()
+    except DataSetError:
+        # We don't care about it for this test, we just want to know that
+        # azure service was called with the right parameters
+        pass
+    blob_service.assert_called_with(
+        account_name="ACCOUNT_NAME", account_key="ACCOUNT_KEY"
+    )
+
+
+@patch("kedro.contrib.io.azure.json_blob.BlockBlobService")
+def test_pass_credentials_save(blob_service, blob_json_data_set, dummy_dataframe):
+    blob_json_data_set().save(dummy_dataframe)
+    blob_service.assert_called_with(
+        account_name="ACCOUNT_NAME", account_key="ACCOUNT_KEY"
+    )
+
+
+@patch("kedro.contrib.io.azure.json_blob.BlockBlobService.get_blob_to_bytes")
+def test_load_blob_args(get_blob_mock, blob_json_data_set):
+    try:
+        blob_json_data_set().load()
+    except DataSetError:
+        pass
+
+    get_blob_mock.assert_called_with(
+        container_name=TEST_CONTAINER_NAME, blob_name=TEST_FILE_NAME, to_extra=42
+    )
+
+
+class BlobMock:
+    def __init__(self):
+        self._content = '{"name": ["tom", "bob"], "age": [3, 4]}\n'.encode("utf-8")
+
+    @property
+    def content(self):
+        return self._content
+
+
+def mock_load_func():
+    def mocked(*args, **kwargs):
+        return BlobMock()
+
+    return mocked
+
+
+@patch(
+    "kedro.contrib.io.azure.json_blob.BlockBlobService.get_blob_to_bytes",
+    new_callable=mock_load_func,
+)
+def test_load(get_blob_mock, blob_json_data_set):
+    result = blob_json_data_set().load()[["name", "age"]]
+    expected = pd.DataFrame({"name": ["tom", "bob"], "age": [3, 4]})
+    expected = expected[["name", "age"]]
+    assert result.equals(expected)
+
+
+class BlobMockDelimited:
+    def __init__(self):
+        self._content = '{"name": "tom", "age": 3}\n{"name": "bob", "age": 4}\n'.encode("utf-8")
+
+    @property
+    def content(self):
+        return self._content
+
+
+def mock_load_delimited_func():
+    def mocked(*args, **kwargs):
+        return BlobMockDelimited()
+
+    return mocked
+
+
+@patch(
+    "kedro.contrib.io.azure.json_blob.BlockBlobService.get_blob_to_bytes",
+    new_callable=mock_load_delimited_func,
+)
+def test_load_delimited(get_blob_mock, blob_json_data_set):
+    result = blob_json_data_set(load_args={"lines": True}).load()[["name", "age"]]
+    expected = pd.DataFrame({"name": ["tom", "bob"], "age": [3, 4]})
+    expected = expected[["name", "age"]]
+    assert result.equals(expected)
+
+
+@patch("kedro.contrib.io.azure.json_blob.BlockBlobService.create_blob_from_bytes")
+def test_save_blob_args(blob_from_mock, blob_json_data_set, dummy_dataframe):
+    blob_json_data_set().save(dummy_dataframe)
+    blob_from_mock.assert_called_with(
+        container_name=TEST_CONTAINER_NAME,
+        blob_name=TEST_FILE_NAME,
+        blob=dummy_dataframe.to_json().encode("utf-8"),
+        from_extra=42,
+    )
+
+
+# pylint: disable=protected-access
+def test_load_extra_params(blob_json_data_set):
```

Comment:
This test seems to be testing that python's assignment operator works in the constructor. Could you get rid of it?